t/db/004_course_users.t

```diff
@@ -161,90 +165,78 @@ my $quimby_params = clone($course_user_params);
+	"addCourseUser: the course doesn't exist"
+);
 
 # updateUser:Check that the user updates.
 my $updated_user = { params => { comment => 'Mayor Joe is the best!!' }, recitation => '2' };
 
-throws_ok {
-	$user_rs->addCourseUser(info => { course_name => 'Arithmetic', username => 'moe' });
-}
-'DB::Exception::UserAlreadyInCourse', 'addCourseUser: the user is already a member';
+is(
+	dies { $user_rs->addCourseUser(info => { course_name => 'Arithmetic', username => 'moe' }); },
+	check_isa('DB::Exception::UserAlreadyInCourse'),
+	'addCourseUser: the user is already a member'
+);
 
 # try to add a non-existent user from a course:
-throws_ok {
-	$user_rs->addCourseUser(info => { course_name => 'Arithmetic', username => 'non_existent_user' })
-}
-'DB::Exception::UserNotFound', 'addCourseUser: try to add a non-existent user to a course';
+is(
+	dies { $user_rs->addCourseUser(info => { course_name => 'Arithmetic', username => 'non_existent_user' }) },
+	check_isa('DB::Exception::UserNotFound'),
+	'addCourseUser: try to add a non-existent user to a course'
+);
 
 # addCourseUser: add a user with undefined parameters
-throws_ok {
-	$user_rs->addCourseUser(
-		info => {
-			course_name => 'Topology',
-			username    => 'quimby',
-		},
-		params => {
-			role            => 'student',
-			undefined_field => 1
-		}
-	);
-}
-'DBIx::Class::Exception', 'addCourseUser: an undefined field is passed in';
+is(
+	dies {
+		$user_rs->addCourseUser(
+			info   => { course_name => 'Topology', username        => 'quimby', },
+			params => { role        => 'student',  undefined_field => 1 }
+		);
+	},
+	check_isa('DBIx::Class::Exception'),
+	'addCourseUser: an undefined field is passed in'
+);
 
 # Add a user with undefined course user parameters.
-throws_ok {
-	$user_rs->addCourseUser(
-		info => {
-			course_name => 'Topology',
-			username    => 'quimby'
-		},
-		params => {
-			role               => 'student',
-			course_user_params => {
-				this_is_not_valid => 1
-			}
-		}
-	);
-}
-'DB::Exception::InvalidField', 'addCourseUser: an undefined parameter is set';
+is(
+	dies {
+		$user_rs->addCourseUser(
+			info   => { course_name => 'Topology', username           => 'quimby' },
+			params => { role        => 'student',  course_user_params => { this_is_not_valid => 1 } }
+		);
+	},
+	check_isa('DB::Exception::InvalidField'),
+	'addCourseUser: an undefined parameter is set'
+);
 
 # Add a user with nonvalid fields
-throws_ok {
-	$user_rs->addCourseUser(
-		info => {
-			course_name => 'Topology',
-			username    => 'quimby'
-		},
-		params => {
-			role               => 'student',
-			course_user_params => {
-				useMathQuill => 0
-			}
-		}
-	);
-}
-'DB::Exception::InvalidParameter', 'addCourseUser: an parameter with invalid value';
+is(
+	dies {
+		$user_rs->addCourseUser(
+			info   => { course_name => 'Topology', username           => 'quimby' },
+			params => { role        => 'student',  course_user_params => { useMathQuill => 0 } }
```

Comment:
useMathQuill is supposed to be a boolean.  Switch to false?

Reply:
This is supposed to be 0.  That is the point of this test.  It is checking that an invalid field is passed in the parameters.  If this is switched to `false` an exception will not be thrown because that is valid.